swagger.yaml

```diff
@@ -2624,7 +2723,7 @@ paths:
         "200":
           description: Success.
           schema:
-            $ref: '#/definitions/supported-operators-list'
+            $ref: '#/definitions/monitored-operators-list'
```

Comment:
My understanding is that the original schema represented operator types that are supported by the installer and the sub-resource (`/supported-operators/{operator_type}`) was meant to return configuration options of a supported operator - both static values corresponding to implemented operator plugins.

This change introduces more elaborate model that for this endpoint will always have only two properties set: `properties` and `operator_type`. Most, if not all other properties are related to a specific cluster state. I think it is a good idea to expose the available configuration options (`properties`) at the listing operation level and IMO it should be a separate model than one used for representing cluster-related operator details; `supported-operators-list` should contain only list of objects containing `properties` ("JSON-formatted string containing description of properties for each operator") and `operator_type` properties.
@YuviGold, @tsorya, @avishayt - what do you think?

Reply:
I've created PR implementing MGMT-4065: #1197, that shows what I had in mind in the comment above.

Reply:
I agree on having separate dedicated objects per endpoint is better than returning the whole object while most of its fields are empty. On the other hand, @tsorya thought it's easier and "flatter" that way :) 
Anyway, I don't care about making more changes if you think it suites better.

Reply:
as we talked previously, we will need to have another object for supported-operators due to the fact it is not part of db. It should be object that is filled (maybe dynamically by going through operators objects) when new supported operator is added. It will not look the same as monitored operator and will have another fields.